### PR TITLE
Moving from GSF to Conscrypt

### PR DESCRIPTION
**Pull request by kaeptmblaubaer1000**
Currently this isn't for merge, just for Travis CI.
I'll write when done.

### CODE REVIEW

**Comment by maks** [comment]
@kaeptmblaubaer1000 thanks for looking into this. I've added #288 to track this work.

**Comment by maks** [comment]
@kaeptmblaubaer1000 please don't merge your feature branch with master, rather the preferred workflow is to stay in sync with master by rebasing your branch onto master.

**Comment by kaeptmblaubaer1000** [comment]
Undone the merge commit by re-importing a patch from the commit after it.

**Comment by kaeptmblaubaer1000** [comment]
Have you disabled travis build?

**Comment by maks** [comment]
@kaeptmblaubaer1000 no, Travis CI should be working as usual

**Comment by kaeptmblaubaer1000** [comment]
Then it's interesting that my commits aren't built.

**Comment by kaeptmblaubaer1000** [comment]
But at least was Travis should test works, just tried with my computer.

**Comment by kaeptmblaubaer1000** [comment]
Now also tested my new conscrypt based SSL implementation.
An idea: (less priority for me) adding a display which implementation is currently used. Anything else here is done.

**Comment by maks** [comment]
Yes I'll need to look at why CI builds are not happening for this PR...
Also can you confirm this PR fixes issues connecting to GitHub over HTTPS on for example Android 4.4 ?

**Comment by kaeptmblaubaer1000** [comment]
No, currently I only tested 4.1.2.
But today I'll test with 4.4.

**Comment by maks** [comment]
Ok, 4.1.2 is fine, just wanted to check it was tested as working on pre 5.0 versions.

File: .travis.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ before_script:
   - export DATE=$(date)
   - export VERSION=$(date --date="$DATE" +'%Y%m%d%H%M%S')
 script:
-  - ./gradlew clean assembleDebug testDebug 
+  - ./gradlew clean build
```
Comment (maks) [inline .travis.yml]: why do the existing gradle targets for CI need to be changed?

File: app/build.gradle [hunk 2]
```diff
@@ -23,6 +23,18 @@ android {
     lintOptions {
         abortOnError false
     }
+
+    flavorDimensions "ssl"
+
+    productFlavors {
```
Comment (maks) [inline app/build.gradle]: I'm happy to remove the dependency on Play services altogether if using Consrcypt directly works. No need to have both.

File: app/build.gradle [hunk 3]
```diff
@@ -44,7 +56,8 @@ dependencies {
     compile libraries.securePreferences
     compile libraries.acra
 
-    compile "com.google.android.gms:play-services-base:11.8.0"
+    playCompile "com.google.android.gms:play-services-base:11.8.0"
```
Comment (maks) [inline app/build.gradle]: see above, if we are going to use consrypt directly then just remove Plat services dep.

File: app/src/main/java/me/sheimi/android/activities/SheimiFragmentActivity.java [hunk 4]
```diff
@@ -1,6 +1,7 @@
 package me.sheimi.android.activities;
 
 import android.app.AlertDialog;
+import android.content.Context;
```
Comment (maks) [inline app/src/main/java/me/sheimi/android/activities/SheimiFragmentActivity.java]: PLease do not include code hanges that are not related to the purpose of this PR (removing dep on Play by using Conscrypt directly).
There is a huge amount of unrelated changes in this PR. Please put them all in seperate PR's, the smaller the better, even single commit PRs are fine.
It might even be easier to abandon and close this PR and create a new one with just the relevant Conscrypt changes.

File: .travis.yml (hunk 1 above)
Comment (kaeptmblaubaer1000) [reply .travis.yml]: Before that change, testDebug wouldn't exist, travis build failed.
It's possible to revert this change if removing product flavors.

File: app/src/main/java/me/sheimi/android/activities/SheimiFragmentActivity.java (hunk 4 above)
Comment (kaeptmblaubaer1000) [reply app/src/main/java/me/sheimi/android/activities/SheimiFragmentActivity.java]: This change is for showGooglePlayError which now isn't in the Activity now.

**Comment by kaeptmblaubaer1000** [comment]
Superseded by #302.
